src/YarpUtilities/src/RosPublisher.cpp

```diff
@@ -0,0 +1,482 @@
+/**
+ * @file RosPublisher.cpp
+ * @authors Prashanth Ramadoss
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include "BipedalLocomotion/YarpUtilities/RosPublisher.h"
+
+#include <yarp/os/Time.h>
+#include <yarp/os/Node.h>
+#include <yarp/os/Publisher.h>
+
+#include <yarp/dev/PolyDriver.h>
+#include <yarp/dev/IFrameTransform.h>
+#include <yarp/sig/Matrix.h>
+
+#include <yarp/rosmsg/TickTime.h>
+#include <yarp/rosmsg/sensor_msgs/JointState.h>
+#include <yarp/rosmsg/geometry_msgs/WrenchStamped.h>
+
+#include <unordered_map>
+#include <algorithm>
+
+using namespace BipedalLocomotion::YarpUtilities;
+using namespace BipedalLocomotion;
+
+template <typename PublisherPtr, typename Msg>
+struct PublisherDetails
+{
+    PublisherPtr ptr{nullptr}; /**< YARP object for a ROS  publisher */
+    std::string topic{"/topic"}; /**< topic over which the message is published */
+    Msg msg; /**< YARP object for a ROS message */    
+};
+
+using JointStateMsg = yarp::rosmsg::sensor_msgs::JointState;
+using JointStatePublisherPtr = std::unique_ptr< yarp::os::Publisher<JointStateMsg> >;
+using WrenchStampedMsg = yarp::rosmsg::geometry_msgs::WrenchStamped;
+using WrenchStampedPublisherPtr = std::unique_ptr< yarp::os::Publisher<WrenchStampedMsg> >;
+using WrenchPublisherDetails = PublisherDetails<WrenchStampedPublisherPtr, WrenchStampedMsg>;
+using JointStatePublisherDetails = PublisherDetails<JointStatePublisherPtr, JointStateMsg>;
+
+class RosPublisher::Impl
+{
+public:
+    /**
+     * Get ROS timestamp from Yarp time now
+     */
+    yarp::rosmsg::TickTime getTimeStampFromYarp();
+        
+    /**
+     * Open the ROS topic for the corresponding publisher
+     * In ROS terms, advertise the topic over which the node is going to publish
+     */
+    template <typename Publisher>
+    bool openPublisher(Publisher* pub, const std::string& topicName)
+    {
+        std::string_view printPrefix = "[RosPublisher::Impl::openPublisher] ";
+        if (!pub->topic(topicName))
+        {
+            std::cerr << printPrefix << "Could not open ROS topic " << topicName << std::endl;
+            return false;
+        }
+        return true;
+    }
+    
+    /**
+     * Utility function to load vector parameters
+     */
+    template<typename Scalar>
+    bool setupParamV(std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler,
+                     const std::string& param, std::vector<Scalar>& vec, const std::string& prefix)
+    {
+        auto handle = handler.lock();
+        if (handle == nullptr)
+        {
+            return false;
+        }
+
+        if (!handle->getParameter(param, GenericContainer::make_vector(vec, GenericContainer::VectorResizeMode::Resizable)))
```

Comment:
By default,
```
getParameter(param, vec)
```
should consider ``vec`` as resizable. The syntax you used is useful when you don't want the vector to be considered resizable.

Reply:
Addressed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/156/commits/d57a286f4780f87c335ff0b3b91ca01443ffa069